cmd/gardener-admission-controller/app/gardener_admission_controller.go

```diff
@@ -0,0 +1,221 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/gardener/gardener/pkg/admissioncontroller/apis/config"
+	admissioncontrollerconfigv1alpha1 "github.com/gardener/gardener/pkg/admissioncontroller/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/admissioncontroller/server/handlers/webhooks"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	clientmapbuilder "github.com/gardener/gardener/pkg/client/kubernetes/clientmap/builder"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/server"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// options has all the context and parameters needed to run a Gardener admission controller.
+type options struct {
+	// ConfigFile is the location of the Gardener controller manager's configuration file.
+	ConfigFile string
```

Comment:
I think, this field can be unexported.
```suggestion
	configFile string
```